test-lit/test_splitter.py

```diff
@@ -0,0 +1,126 @@
+""" test_splitter.py
+
+    Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+    See https://llvm.org/LICENSE.txt for license information.
+    SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+    
+"""
+
+import argparse
+import os
+import re
+from pathlib import Path
+
+
+class TestSplitter:
+    def __init__(self):
+        self.error_tests = []
+        self.modfile_tests = []
+        self.symbol_tests = []
+        self.generic_tests = []
+        self.unsupported_tests = []
+        self.test_types = ['ERROR', 'MODFILE', 'SYMBOL', 'GENERIC']
+        root = os.getcwd()
+        while not root.endswith("f18"):
+            root = os.path.abspath(os.path.dirname(root))
+        self.root = root
+        self.test_path = os.path.join(self.root, "test")
+        self.get_semantics_tests()
+        self.tests = (self.error_tests + self.modfile_tests +
+                      self.symbol_tests + self.generic_tests +
+                      self.unsupported_tests)
+
+    def get_semantics_tests(self):
+        semantics_path = os.path.join(self.test_path, "semantics")
+        semantics_cmake_path = os.path.join(semantics_path, "CMakeLists.txt")
+        with open(semantics_cmake_path, "r", encoding="utf8") as read_file:
+            current_test_set = None
+            for line in read_file.readlines():
+                match = re.match(r"^set\((.*)$", line)
+                if match:
+                    if match.group(1) == "ERROR_TESTS":
+                        current_test_set = self.error_tests
+                    elif match.group(1) == "MODFILE_TESTS":
+                        current_test_set = self.modfile_tests
+                    elif match.group(1) == "SYMBOL_TESTS":
+                        current_test_set = self.symbol_tests
+                    else:
+                        current_test_set = self.generic_tests
+                elif re.match(r".*\..*f[90|\]| \n]", line):
```

Comment:
I believe that we could have other files, no? Will this only work if the files are f.90?

Reply:
The regex matches every entry in the CMakeLists.txt which is its purpose. It includes things like `  canondo*.[Ff]90`, but is globbed to get all files after matching.